pkg/controllers/webhook/controller.go

```diff
@@ -738,6 +746,10 @@ func (c *controller) buildResourceValidatingWebhookConfiguration(caBundle []byte
 		if len(webhookCfgs) > 0 {
 			webhookCfg = webhookCfgs[0]
 		}
+		sideEffects := &none
+		if c.admissionReports {
+			sideEffects = &noneOnDryRun
```

Comment:
Can you help me better understand this change? Why have different sideEffects based on the admissionReports setting?

Reply:
When not creating admission reports, invoking has no side effect.
As soon as we start creating something when the webhook is invoked it's a side effect.
Does it make sense ?